src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadAndUpdateOperation.java

```diff
@@ -168,7 +170,7 @@ private Uni<UpdatedDocument> processUpdate(
                       v -> {
                         if (readDocument.txnId() != null) modifiedCount.incrementAndGet();
                         return new UpdatedDocument(
-                            readDocument.id(),
+                            writableShreddedDocument.id(),
```

Comment:
So how is this a fix and wouldn't this end up in the `NPE` in that else case? Something is off here..

Reply:
By default updated Document id (updated) is set as Uni.nullValue() when writableShreddedDocument is null. This handled in the below code as 
updated
                  .onItem()
                  **.ifNotNull()**